### PR TITLE
Keycloak usage scenarios documentation

### PR DESCRIPTION
**Pull request by arso**
Signed-off-by: Artur Socha <asocha@redhat.com>

### CODE REVIEW

File: keycloak_usage.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+[ INFO  ] Generating answer file '/var/lib/ovirt-engine/setup/answers/20220519112026-setup.conf'                                     
+[ INFO  ] Stage: Pre-termination                                                                                                     
+[ INFO  ] Stage: Termination                                                           
+[ INFO  ] Execution of setup completed successfully                                                                                  
+```
+
+# Current limitations
+
+## Fallback
+There is no automated fallback script from internal Keycloak to legacy AAA modules. However, this can be done manually if needed but it is not recommended
```
Comment (didib) [inline keycloak_usage.md:96]: ... but is not recommended.

**Review by didib — APPROVED** [review]

File: keycloak_usage.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          * Please note * : Keycloak is now deprecating AAA/JDBC authentication module.
+          It is highly recommended to install Keycloak based authentication.
+          Configure Keycloak on this host (Yes, No) [Yes]:
+          provided answer: no
+
+          Are you really sure not to install internal Keycloak based authentication?
+          AAA modules are being deprecated
+          Configure Keycloak on this host (Yes, No) [Yes]:
+          provided answer: no
+```
```
[the remark this answers is not in the document]
Comment (arso) [reply keycloak_usage.md:27]: Added/

File: keycloak_usage.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+## Reconfiguration from oVirt 4.5.1 and above
+
+In case when system administrator originally disabled internal Keycloak for oVirt >= 4.5.1 and later decides to make a switch it is possible to do so. The trade off is that any existing AAA configuration will have to be migrated to Keycloak manually.
```
[the remark this answers is not in the document]
Comment (arso) [reply keycloak_usage.md]: I have updated it according to Didi's recommendation.

File: keycloak_usage.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+In order to activate internal Keycloak please see the [activation procedure](#Internal-Keycloak-activation-procedure).
```
[the remark this answers is not in the document]
Comment (arso) [reply keycloak_usage.md]: Fixed

File: keycloak_usage.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+In case system administrator decides to enable internally bundled Keycloak SSO there are the following steps needed:
+1. Backup existing setup (full backup & restore exercise is highly recommended)
+2. Enable internally bundled Keycloak. Please note that any existing user base setup needs to be migrated manually. Please see Keycloak documentation how to setup various user backends (todo: links)
+```
+engine-setup --otopi-environment="OVESETUP_KEYCLOAK_CONFIG/adminPassword=str:CHANGE_ME \
+ OVESETUP_KEYCLOAK_CORE/enable=bool:True \
+ OVESETUP_KEYCLOAK_DB/newDatabase=bool:True" `
+```
+3. To test the setup please login to AdministratorPanel using the admin username: `admin@ovirt` and provided password. For REST API access the full username with profile must be provided: `admin@ovirt@internalsso`. In order to login to Keycloak Administration Console please use `admin` and provided password (from the above).
```
[the remark this answers is not in the document]
Comment (arso) [reply keycloak_usage.md]: Fixed

File: keycloak_usage.md (hunk 1 above)
Comment (arso) [reply keycloak_usage.md]: Fixed

File: keycloak_usage.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+In order to activate internal Keycloak please see the [activation procedure](#Internal-Keycloak-activation-procedure).
+
+
+## Upgrade from existing oVirt Engine installations (oVirt <= 4.5.0)
+
+When upgrading the existing oVirt installations(to oVirt >= 4.5.1), due to backward compatibility, oVirt AAA configuration is the preferred solution. There will be no Keycloak related questions during engine-setup execution. System administrators can decide, whether to migrate manually as documented [here](https://www.ovirt.org/documentation/administration_guide/index.html#Configuring_Red_Hat_SSO), or keep using their existing setup. At current time there are no automatic migration from AAA JDBC or AAA LDAP to internally bundled Keycloak.
+In order to activate internal Keycloak please see the [activation procedure](#Internal-Keycloak-activation-procedure).
+
+## Development environment
+Keycloak is not supposed to be installed and configured on development environment.
```
[the remark this answers is not in the document]
Comment (arso) [reply keycloak_usage.md]: fixed

[the remark this answers is not in the document]
Comment (didib) [reply keycloak_usage.md]: Perhaps link?

[the remark this answers is not in the document]
Comment (arso) [reply keycloak_usage.md]: ah - right!

File: keycloak_usage.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (arso) [reply keycloak_usage.md]: fixed again - thanks!

**Review by mwperina — APPROVED** [review]
+1

**Pull request merged**